advplyr/audiobookshelf
# Improve RSS Manager adding Incoming Feed Health Check

This pull request introduces a new feature that improves podcast management by allowing users to check the health status of feed URLs for incoming podcasts:

1. **Feed Health Status**: Users can now easily determine whether a podcast's feed URL is healthy or not. Every time the AudioBookShelf accesses the feed URL, it performs a verification and marks the feed as either healthy or not.

2. **Manager Page**: A new tab has been added on the Confi -> RSS Feeds, where users can view all the podcasts that contain a feed URL. This screen provides the following information:

   - **Podcast Title**
   - **Feed URL**: The Feed URL. It's truncated because sometimes the URL is gigantic
   - **Last Checked**: It's the last date and time that the Feed URL was checked
   - **Last Successful Check Date** (new table column): Displays the date and time of the application able to access the feed URL with success
   - **Feed Healthy** (new table column): Indicates whether the feed URL is marked as healthy or not.
   - **Next Automatic Check**: If the feed has a cron, show the next time that the feed will be checked
   - **Force Recheck**: Allows users to manually trigger a recheck of the feed's health status.

**Reason for the Changes**

This feature was added to provide users with greater transparency and control over the health of incoming podcast feed URLs. It empowers users to make informed decisions about which feeds to subscribe to based on their health status.

**How to Test**

To test this feature, follow these steps:

1. Navigate to the Config --> RSS Feeds.
2. Check the list of podcasts with associated feed URLs.
2. Click the "Force Recheck" button for a podcast to trigger a manual recheck.
3. Verify that the health status is updated accordingly.

**Additional Notes**

- Every time AudioBookShelf accesses the feed, it saves the feed's healthy status (e.g., for downloading new episodes or displaying them on the Podcast Detail page).


**What's missing to finish**

- [ ] The podcast cover arts are not being displayed on the page.

- [ ] I attempted to use the Podcast model, but I am unsure whether I should use the LibraryItem old method or the Podcast Model. As a result, when I execute the following code:

```
await Database.podcastModel.update(podcast, {
 where: { id: podcastId }
 })
```

The update is temporarily saved, but it does not appear to be saved in the database.

-----

The new Page working

https://github.com/advplyr/audiobookshelf/assets/814828/ea3e45af-7ff2-4bcb-a93a-7b947e2a834b

-----

Subscribed Feeds page

![image](https://github.com/advplyr/audiobookshelf/assets/814828/1134ac35-75ab-4f1e-bd53-3aca2f9dda99)

-----

The RSS Page has been renamed to Opened Feeds and now it's a tab on the RSS Page, that includes a new search box to filter by title.

![image](https://github.com/advplyr/audiobookshelf/assets/814828/6d69bb7e-e9b3-4183-a74d-8d9a3b5e4693)

-----

Podcast health icon displayed with feed URL and date of last update on details page:

With a healthy feed
<img width="817" alt="Screenshot 2023-10-03 at 01 25 09" src="https://github.com/advplyr/audiobookshelf/assets/814828/77a19ac1-7718-4498-b36f-ee3964aa7cc1">

With an unhealthy feed
<img width="817" alt="Screenshot 2023-10-03 at 01 31 12" src="https://github.com/advplyr/audiobookshelf/assets/814828/b76d6f89-f9a4-4f60-9ee9-1e13547ccef3">

Without a feed URL
<img width="817" alt="Screenshot 2023-10-03 at 01 30 02" src="https://github.com/advplyr/audiobookshelf/assets/814828/ec4c1a4e-91b4-4897-b492-9d5cb0d443e2">
